### ISSUE TITLE
Make `status dirname` work correctly in a script after `$PWD` changes

### ISSUE THREAD

**Issue by andrey-mishchenko** [fish-shell/fish-shell#8562]
```
~/tmp $ cat test.fish 
status dirname
cd ..
status dirname
~/tmp $ fish test.fish
.
.
~/tmp $ 
```

What do you think about making the second `status dirname` call return `tmp`, so that the mental model of `status dirname` is that at any point in a script, it returns the path to the folder containing that script?

**Comment by floam** [comment]
I don't think so, it's specifically intended to provide you the path of the script you're running, not $PWD, I think?

You can use $PWD or `pwd` to get the current directory.

**Comment by faho** [comment]
>I don't think so, it's specifically intended to provide you the path of the script you're running, not $PWD, I think?

It is, yes. It's short for `dirname (status filename)`.

Only it's not absolutized, so if you start a script in the current directory it will effectively do `dirname test.fish`, which doesn't have a directory, and so it will just print `.`. Which, of course, isn't useful after you've changed directory anymore.

There is a point to be made, there. Unfortunately it appears tricky to add this, because we don't currently store the script's directory anywhere, only the (relative) filename. So once you've cd'd, we've forgotten what it refers to to begin with.


It's easy enough to work around:

At the beginning of the script, run `set -l dir (builtin realpath (status dirname))`. Then use that whenever you need the dir.

(also ask yourself whether you need the script's directory to begin with - in my experience many people do this reflexively but never actually need it)

**Comment by andrey-mishchenko** [comment]
> I don't think so, it's specifically intended to provide you the path of the script you're running, not $PWD, I think?
> 
> You can use $PWD or `pwd` to get the current directory.
